ckan/lib/plugins.py

```diff
@@ -198,54 +200,57 @@ def register_group_plugins(map):
             # routing setup
-
-            map.connect('%s_index' % group_type, '/%s' % group_type,
-                        controller=group_controller, action='index')
-            map.connect('%s_new' % group_type, '/%s/new' % group_type,
-                        controller=group_controller, action='new')
-            map.connect('%s_read' % group_type, '/%s/{id}' % group_type,
-                        controller=group_controller, action='read')
-            map.connect('%s_action' % group_type,
-                        '/%s/{action}/{id}' % group_type,
-                        controller=group_controller,
-                        requirements=dict(action='|'.join(
-                            ['edit', 'authz', 'delete', 'history', 'member_new',
-                             'member_delete', 'followers', 'follow',
-                             'unfollow', 'admins', 'activity'])))
-            map.connect('%s_edit' % group_type, '/%s/edit/{id}' % group_type,
-                        controller=group_controller, action='edit',
-                        ckan_icon='pencil-square-o')
-            map.connect('%s_members' % group_type,
-                        '/%s/members/{id}' % group_type,
-                        controller=group_controller,
-                        action='members',
-                        ckan_icon='users')
-            map.connect('%s_member_new' % group_type,
-                        '/%s/member_new/{id}' % group_type,
-                        controller=group_controller,
-                        action='member_new')
-            map.connect('%s_member_delete' % group_type,
-                        '/%s/member_delete/{id}' % group_type,
-                        controller=group_controller,
-                        action='member_delete')
-            map.connect('%s_activity' % group_type,
-                        '/%s/activity/{id}/{offset}' % group_type,
-                        controller=group_controller,
-                        action='activity', ckan_icon='clock-o'),
-            map.connect('%s_about' % group_type, '/%s/about/{id}' % group_type,
-                        controller=group_controller,
-                        action='about', ckan_icon='info-circle')
-            map.connect('%s_bulk_process' % group_type,
-                        '/%s/bulk_process/{id}' % group_type,
-                        controller=group_controller,
-                        action='bulk_process', ckan_icon='sitemap')
+            blueprint = Blueprint(group_type, group.group.import_name,
+                                  url_prefix='/{}'.format(group_type),
+                                  url_defaults={'package_type': group_type})
+            actions = ['group.member_delete', 'group.history', 
+                       'group.followers', 'group.follow', 'group.unfollow',
+                       'group.admins', 'group.activity']
+
+            blueprint.add_url_rule(u'/', view_func=group.index,
```

Comment:
adding `url_defaults` means that this keywords will be passed to each function registered under blueprint. i.e, `group.index` will be called as group_index(package_type), `group.read` - read(package_type, id), not just read(id). You can check my `dataset` file and you'll see, that all views are expecting this argument, even though it doesn't appears in the rules.